### PR TITLE
Kyle transaction and invoice refactor

### PR DESCRIPTION
**Pull request by Kledin85**

### CODE REVIEW

**Comment by KaraJoHo** [comment]
Nice!

**Review by KaraJoHo — APPROVED** [review]

File: spec/invoice_repository_spec.rb [hunk 1]
```diff
@@ -8,17 +8,60 @@
...
+      )
+
+      ivr.all.push(i1,i2,i3)
+
+      ivr.create(
+        :id          => 7,
+        :customer_id => 9,
+        :merchant_id => 10,
+        :status      => 'completed',
+        :created_at  => Time.now.to_s,
```
[the remark this answers is not in the document]
Comment (KaraJoHo) [reply spec/invoice_repository_spec.rb:54]: Think we're okay to leave this

**Review by beddings81 — APPROVED** [review]
Great job!

**Review by Pocketzs — APPROVED** [review]
No changes needed from me

**Pull request merged**